spec/models/profile_spec.rb

```diff
@@ -288,6 +288,30 @@
     end
   end
 
+  describe "#has_proofed_before" do
+    it 'is false when the user has only been activated once' do
+      profile.activate
+      expect(profile.has_proofed_before?).to be_falsey
+    end
+
+    it 'is true when the user is re-activated' do
+      existing_profile = Profile.create(user: user)
+      existing_profile.activate
+      profile.activate
+
+      existing_profile.reload
+      profile.reload
+
+      # Now, existing_profile should be deactivated
+      expect(existing_profile.active).to be_falsey
+      expect(existing_profile.activated_at).to_not be_nil
+      expect(profile.active).to be_truthy
+      expect(profile.activated_at).to_not be_nil
```

Comment:
This little block is probably unnecessary, but was useful for me proving out the concept. Would you like me to remove 305-309?

Reply:
seems ok to keep for now